compiler/.clang-format

```diff
@@ -0,0 +1,169 @@
+###############################################################################
+# Copyright IBM Corp. and others 2015
```

Comment:
Didn't realize the copyright date was incorrect; I will update this when I rebase on the latest master when we enter the code freeze.